pkgs/tools/networking/zerotierone/default.nix

```diff
@@ -65,6 +65,19 @@ in stdenv.mkDerivation {
 
   enableParallelBuilding = true;
 
+  # Ensure Rust compiles for the right target
+  env.CARGO_BUILD_TARGET = stdenv.hostPlatform.rust.rustcTarget;
+
+  # Cargo won't compile to target/release but to target/<RUST_TARGET>/release when a target is
+  # explicitly defined. The build-system however expects target/release. Hence we just symlink from
+  # the latter to the former.
+  preBuild = ''
+    mkdir --parent -- rustybits/target/release
+    ln --relative --symbolic -- \
+      $PWD/rustybits/target/${stdenv.hostPlatform.rust.rustcTarget}/release/libzeroidc.a \
```

Comment:
$PWD can be replaced by .

Reply:
Unfortunately not. The `TARGET` in `ln TARGET LINK_NAME` does not exist at the point in time when creating the symlink. Hence `ln` does no magic in resolving things, and we end up with a relative symlink inside `rustybits/target/release` that points to (no leading symbols!) `rustybits/target/${stdenv.hostPlatform.rust.rustcTarget}/release/libzeroidc.a`, when instead it should point to `../../../rustybits/target/${stdenv.hostPlatform.rust.rustcTarget}/release/libzeroidc.a`.

Now why the `--relative`? So that this step works even when not building instide `/build`, e.g. when using a nix-shell!